js/super_hero.js

```diff
@@ -1 +1,31 @@
-// var SuperHero = ...
+
+var SuperHero = function(name, power){
+    this.name = name;
+    this.power = power;
+    this.hp = 100;
+};
+
+SuperHero.prototype = {
+    setPower: function(power){
+        this.power = power;
+    },
+    setName: function(name){
+        this.name = name;
+    },
+    getHit: function(amt){
+        this.hp -= amt;
+        console.log(this.name+" hit!");
+        console.log(this.name+" HP: "+this.hp);
+    },
+    attack: function(other, amt){
+        console.log(this.name+" uses "+this.power+" against "+other.name+"!");
+        other.getHit(amt);
+    }
+};
+
+var vegeta = new SuperHero(name="Vegeta", power="Galick Gun");
```

Comment:
FYI, there's an (presumably accidental) side effect in this line...can you spot what it is? I'll give you a hint: global variables.